sycl/include/CL/sycl/detail/pi.hpp

```diff
@@ -215,7 +215,13 @@ template <> inline void print<>(PiResult val) {
 }
 
 // cout does not resolve a nullptr.
-template <> inline void print<>(std::nullptr_t val) { print<void *>(val); }
+template <> inline void print<>(std::nullptr_t val) {
+  std::cout << "<nullptr> : " << static_cast<void *>(val) << std::endl;
+}
+
+template <> inline void print(char *val) {
+  std::cout << "<char * > : " << static_cast<void *>(val) << std::endl;
```

Comment:
What was printed before for `char *` and how is this better? Can't we really print the C-string instead of the pointer?

Reply:
In my test case, this is an output variable.   We were printing the contents of memory, which had not been initialized. 
The other way to do this would be to always initialize memory after allocation.

Reply:
Then can we add specialization to print the C-string passed as `const char *`, and only print pointer for `char *`?